browser_history/generic.py

```diff
@@ -175,56 +189,95 @@ def fetch(self, history_paths=None, sort=True, desc=False):
+        """Returns bookmarks of all available profiles stored in SQL or JSON or plist.
+
+        The returned datetimes are timezone-aware with the local timezone set by default.
+
+        The bookmark files are first copied to a temporary location and then queried, this might
+        lead to some additional overhead and results returned might not be the latest if the
+        browser is in use. This is done because the SQlite files are locked by the browser when
+        in use.
+
+        :param bookmarks_paths: (optional) a list of bookmark files.
+        :type bookmarks_paths: list(:py:class:`pathlib.Path`)
+        :param sort: (optional) flag to specify if the output should be sorted.
+            Default value set to True.
+        :type sort: boolean
+        :param desc: (optional)  flag to speicify asc/desc (Applicable iff sort is True)
+            Default value set to False.
+        :type asc: boolean
+        :return: Object of class :py:class:`browser_history.generic.Outputs` with the
+            data member entries set to list(tuple(:py:class:`datetime.datetime`, str))
+        :rtype: :py:class:`browser_history.generic.Outputs`
+        """
+
+        if bookmarks_paths is None:
+            bookmarks_paths = self.paths(profile_file=self.bookmarks_file)
+        output_object = Outputs(fetch_type="bookmarks")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for bookmarks_path in bookmarks_paths:
+                if not os.path.exists(bookmarks_path):
+                    continue
+                copied_bookmark_path = shutil.copy2(
+                    bookmarks_path.absolute(), tmpdirname
+                )
+                date_bookmarks = self.bookmarks_parser(copied_bookmark_path)
+                output_object.bookmarks.extend(date_bookmarks)
+                if sort:
+                    output_object.bookmarks.sort(reverse=desc)
         return output_object
 
 
 class Outputs:
     """
-    A generic class to encapsulate history outputs and to
+    A generic class to encapsulate history and bookmark outputs and to
     easily convert them to JSON, CSV or other formats.
 
-    * **entries**: List of tuples of Timestamp & URL
-    :type entries: list(tuple(:py:class:`datetime.datetime`, str))
+    * **histories**: List of tuples of Timestamp & URL
+    :type history_entries: list(tuple(:py:class:`datetime.datetime`, str))
 
-    * **formats**: A tuple of strings containing all supported formats
+    * **bookmarks** List of tuples of Timestamp , URL , Title , Folder
+    :type bookmark_entries: list(tuple(:py:class:`datetime.datetime`, str,str,str))
 
-    * **fields**: The fields available for the history data returned
+    * **formats**: A tuple of strings containing all supported formats
 
     """
 
     # All formats added here should be implemented in _format_map
     # Formats added here and in _format_map should be in lowercase
     formats = ("csv", "json")
```

Comment:
Should also have `jsonl`. Or just set this to `self._format_map.keys()`.

Reply:
This was from another PR i think I didnt change this, Ill do it if needed

Reply:
Ah right. I think it's better to change it since we'll be specifying all the formats in two places and one might not be updated (like in this case `jsonl` is missing and so it doesn't show up in the CLI help.)